### ISSUE TITLE
[Feature request] Add history to "Hide articles after reading"

### ISSUE THREAD

**Issue by Jucgshu** [FreshRSS/FreshRSS#1695]
Hi Fresh Guys,

I love using the "**Hide articles after reading**" feature of FreshRSS, as it allows me to quickly read my stuff and forget about it when I'm done.

Only sometimes, I press the "J" button (next article) a little bit too quikly, and I want to find back the article I just marked as read. Because I use the "Hide articles after reading", I have to:
1. Use the search feature
2. Find my article back
3. Mark it as unread
4. Find it back in my reading list again

What would be cool is to **have an option to select the number of articles to keep before hiding**, so that even if the articles are marked as read, one can find his read articles quickly.

Thanks for all of the work you're doing guys. FreshRSS is really awesome, and I'm using it every day!

**Comment by Frenzie** [comment]
So you're saying <kbd>k</kbd>/back doesn't work in that case? To me the default is already plenty hide-y, what with only showing unread blog posts and all. :-P

**Comment by Jucgshu** [comment]
Actually <kbd>J</kbd> or <kbd>K</kbd> works in all cases.

*But*.
  
* When "Hide articles after reading" is **not** selected (default), you can <kbd>J</kbd> or <kbd>K</kbd> (```prev``` / ```next```) to navigate through the articles history. But **all** articles stay there until FreshRSS is refreshed (which can be a problem when you navigate through literally thousands of articles).
 * When "Hide articles after reading" **is** selected, it is also possible to hit <kbd>J</kbd> or <kbd>K</kbd> but:
    - <kbd>J</kbd> will navigate to the next article (which should be unread in all cases).
    - <kbd>K</kbd> will navigate to the previous **unread** article, i.e. not the previous (read) one as it has been already marked as read (and hidden by the "Hide articles after reading" option). 

I know this request sounds strange but it's (reasonably) painful to have to go through the search feature when you have marked an article as read and want to find it back. So being able to rewind a bit through read articles would be really nice, even if the "Hide articles after reading" option has been selected.

-j

**Comment by Frenzie** [comment]
> But all articles stay there until FreshRSS is refreshed (which can be a problem when you navigate through literally thousands of articles).

Could you explain how that's a problem exactly? Just refresh it on purpose every once in a while. :-P

I should probably point out that I'm still not completely sold on the way FreshRSS works compared to a traditional feedreader like Opera/Presto, QuiteRSS or Liferea, where only stuff that I explicitly "delete" disappears. I've adapted to using favorites and saving links more in external tools. (I decided to switch over completely nearly two years ago, mind you; I'm just explaining where I'm coming from.)

Anyway, I definitely think it's an interesting proposition for much the same reason. Sometimes I accidentally mark something as read and I have to retrieve it.

**Comment by Jucgshu** [comment]
> Could you explain how that's a problem exactly? Just refresh it on purpose every once in a while. :-P

I mean, I could. But it's would be a UX breakage to me, as it would make me reload the app, instead of a straightforward experience while staying inside FreshRSS the whole time  :wink:

**Comment by Alkarex** [comment]
Currently, we either keep 0 or all articles when they are marked as read. We could merge those two features into a single option allowing to chose how many read articles to keep while reading / scrolling down.

**Comment by Jucgshu** [comment]
Yes, that would make even more sense, while being more user friendly in the settings panel.

**Comment by Alkarex** [comment]
Will be in 1.14.0. Made easy by the refactoring of https://github.com/FreshRSS/FreshRSS/pull/2199
